packages/dataviews/src/view-list.tsx

```diff
@@ -192,6 +369,24 @@ export default function ViewList( props: ListViewProps ) {
 		defaultActiveId: getItemDomId( selectedItem ),
 	} );
 
+	// Manage focused item, when the active one is removed from the list.
+	const isActiveIdInList = store.useState(
+		( state: { items: any[]; activeId: any } ) =>
+			state.items.some(
+				( item: { id: any } ) => item.id === state.activeId
+			)
+	);
+	useEffect( () => {
+		if ( ! isActiveIdInList ) {
+			// Prefer going down, except if there is no item below (last item), then go up (last item in list).
+			if ( store.down() ) {
+				store.move( store.down() );
+			} else if ( store.up() ) {
+				store.move( store.up() );
+			}
+		}
+	}, [ isActiveIdInList ] );
```

Comment:
This is a very weird effect, why do we have this adhoc logic in our view?

Reply:
It fixes [this behavior](https://github.com/WordPress/gutenberg/pull/60805#issuecomment-2107561972) (see related ariakit issue).

Reply:
cc @WordPress/gutenberg-components for awareness. This is another of those things related to Composite that you may want to look at to smooth the rough edges.